core/src/main/scala/org/bitcoins/core/psbt/PSBTHelper.scala

```diff
@@ -0,0 +1,256 @@
+package org.bitcoins.core.psbt
+
+import org.bitcoins.core.crypto.{ECDigitalSignature, ECPublicKey, ExtPublicKey}
+import org.bitcoins.core.hd.BIP32Path
+import org.bitcoins.core.number.UInt32
+import org.bitcoins.core.protocol.CompactSizeUInt
+import org.bitcoins.core.protocol.script.{
+  ScriptPubKey,
+  ScriptSignature,
+  WitnessScriptPubKey
+}
+import org.bitcoins.core.protocol.transaction.{Transaction, TransactionOutput}
+import org.bitcoins.core.psbt.GlobalPSBTRecord._
+import org.bitcoins.core.psbt.InputPSBTRecord._
+import org.bitcoins.core.psbt.PSBTGlobalKey._
+import org.bitcoins.core.psbt.PSBTInputKey._
+import org.bitcoins.core.script.crypto.HashType
+import org.bitcoins.core.serializers.script.RawScriptWitnessParser
+import org.bitcoins.core.util.BitcoinScriptUtil
+import scodec.bits.ByteVector
+
+import scala.annotation.tailrec
+
+object PSBTHelper {
+
+  final private val terminator: Byte = 0x00.byteValue
+
+  private def parseKeyAndValue(
+      bytes: ByteVector): (Long, ByteVector, ByteVector) = {
+    val keyCmpctUInt = CompactSizeUInt.parse(bytes)
+
+    if (keyCmpctUInt.toLong == 0) {
+      (keyCmpctUInt.size, ByteVector.empty, ByteVector.empty)
+    } else {
+      val key = bytes.drop(keyCmpctUInt.size).take(keyCmpctUInt.toLong)
+      val valueBytes = bytes.drop(keyCmpctUInt.size + keyCmpctUInt.toLong)
+      val valueCmpctUInt = CompactSizeUInt.parse(valueBytes)
+      val value = valueBytes
+        .drop(valueCmpctUInt.size)
+        .take(valueCmpctUInt.toLong)
+      val totalSize = keyCmpctUInt.size + keyCmpctUInt.toLong + valueCmpctUInt.size + valueCmpctUInt
+        .toLong
+
+      (totalSize, key, value)
+    }
+  }
+
+  @tailrec
+  def parseGlobalMap(
```

Comment:
For every single `@tailrec` function in this file, I think it would be nice to have a `def` to make the interface nicer for users that didn't have all of the params that aren't necessary to the internal workings of the recursion. For example, this function should have a
```
def parseGlobalMap(bytes: ByteVector) = {
  parseGlobalMap(remainingBytes = bytes, accum = Vector.empty, txOpt = None)
} 
```